includes/Classifai/Providers/Azure/ComputerVision.php

```diff
@@ -405,7 +405,7 @@ public function maybe_rescan_image( $attachment_id ) {
 			$image_url = get_largest_acceptable_image_url(
 				get_attached_file( $attachment_id ),
 				wp_get_attachment_url( $attachment_id ),
-				$metadata['sizes'],
+				wp_attachment_is_image( $attachment_id ) ? $metadata['sizes'] : [],
```

Comment:
So this fixes the first issue but does introduce a new issue. If you look at the `get_largest_acceptable_image_url` helper function, it checks to see if the size of the file is greater than our maximum (4MB by default here). If we're under that, this all works as expected. But if we're over that size, that function then tries to find the largest acceptable size.

In this case, if we pass in an empty array, it can't find any sizes and ends up returning `null`, which breaks other things. So I think along with this fix above, we'll need a fix in the `get_largest_acceptable_image_url` function to return the full image size if our `$sizes` array is empty

Reply:
@dkotter, can we just move this part:
`if ( filter_input( INPUT_POST, 'rescan-pdf', FILTER_SANITIZE_STRING ) ) {
	$this->read_pdf( $attachment_id );
}`
to the top, and exit right after we do `read_pdf`? That way we do the PDF processing early and exit, or continue on to process image-related stuff.

Reply:
Yeah, that's a good approach as well. I'd probably move this whole section up:

```php
// Are we updating the OCR text?
if ( filter_input( INPUT_POST, 'rescan-ocr', FILTER_SANITIZE_STRING ) ) {
	$this->ocr_processing( wp_get_attachment_metadata( $attachment_id ), $attachment_id, true );
}
if ( filter_input( INPUT_POST, 'rescan-pdf', FILTER_SANITIZE_STRING ) ) {
	$this->read_pdf( $attachment_id );
}
```

Will just need to verify that we currently only use the `maybe_rescan_image` method with a single route if we change this to return early. The way it's architectured, it makes it seem like it can be used for multiple routes at once but I'm not aware of anywhere in the current codebase that we do this.